ga4gh/datamodel/reads.py

```diff
@@ -13,6 +13,68 @@
 import pysam
 
 import ga4gh.protocol as protocol
+import ga4gh.exceptions as exceptions
+
+
+class SamCigar(object):
+    """
+    Utility class for working with SAM CIGAR strings
+    """
+    # see http://pysam.readthedocs.org/en/latest/api.html
+    # #pysam.AlignedSegment.cigartuples
+    cigarMap = {
+        protocol.GACigarOperation.ALIGNMENT_MATCH: 0,
+        protocol.GACigarOperation.INSERT: 1,
+        protocol.GACigarOperation.DELETE: 2,
+        protocol.GACigarOperation.SKIP: 3,
+        protocol.GACigarOperation.CLIP_SOFT: 4,
+        protocol.GACigarOperation.CLIP_HARD: 5,
+        protocol.GACigarOperation.PAD: 6,
+        protocol.GACigarOperation.SEQUENCE_MATCH: 7,
+        protocol.GACigarOperation.SEQUENCE_MISMATCH: 8,
+    }
+
+    reverseCigarMap = {
+        0: protocol.GACigarOperation.ALIGNMENT_MATCH,
+        1: protocol.GACigarOperation.INSERT,
+        2: protocol.GACigarOperation.DELETE,
+        3: protocol.GACigarOperation.SKIP,
+        4: protocol.GACigarOperation.CLIP_SOFT,
+        5: protocol.GACigarOperation.CLIP_HARD,
+        6: protocol.GACigarOperation.PAD,
+        7: protocol.GACigarOperation.SEQUENCE_MATCH,
+        8: protocol.GACigarOperation.SEQUENCE_MISMATCH,
+    }
+
+    @classmethod
+    def ga2int(cls, value):
+        return cls.cigarMap[value]
+
+    @classmethod
+    def int2ga(cls, value):
+        return cls.reverseCigarMap[value]
```

Comment:
Huh? We're making a lot of assumptions that may not be future-compatible with changes to the spec (think all those GA prefixes :)) Postulating that a 1-1 invertible map stays that way seems about as safe an assumption to make as can be, and not hand-coding both directions seems like a less error-prone approach in general.